### ISSUE TITLE
PSScriptAnalyzer fails on DSC configuration file on macOS - Unable to load DLL 'libmi.so'

### ISSUE THREAD

**Issue by matt-richardson** [PowerShell/PowerShell#5707]
Steps to reproduce
------------------

```powershell
git clone https://github.com/OctopusDeploy/OctopusDSC.git
cd OctopusDSC
git checkout ce3f797347bec9c0366c33c9b6c3f3e6ddeef6f3
cd OctopusDSC/Examples
Invoke-ScriptAnalyzer -Path ./cOctopusSeqLogger.ps1 
```

Expected behavior
-----------------

Normal script analysis 

Actual behavior
---------------

```none
PS /Users/me/OctopusDSC> Invoke-ScriptAnalyzer ./OctopusDSC/Examples/cOctopusSeqLogger.ps1                                                                                       
Invoke-ScriptAnalyzer : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  System.DllNotFoundException: Unable to load DLL 'libmi.so': The specified module or one of its dependencies could not be found.
 (Exception from HRESULT: 0x8007007E)
   at Microsoft.Management.Infrastructure.Native.NativeMethods.MI_Application_InitializeV1(UInt32 flags, String applicationID, IndirectPtr extendedError, DirectPtr application)
   at Microsoft.Management.Infrastructure.Native.MI_Application.Initialize(String applicationId, MI_Instance& extendedError, MI_Application& application)
   at Microsoft.Management.Infrastructure.Internal.CimApplication.GetApplicationHandle()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.CreateValue()
   at Microsoft.Management.Infrastructure.Internal.CimApplication.get_Handle()
   at Microsoft.Management.Infrastructure.Serialization.CimMofDeserializer..ctor()
   at Microsoft.Management.Infrastructure.Serialization.CimMofDeserializer.Create()
   at Microsoft.PowerShell.DesiredStateConfiguration.CimDSCParser..ctor(OnClassNeeded onClassNeeded)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.ImportClasses(String path, Tuple`2 moduleInfo, Collection`1 errors)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.Initialize(Collection`1 errors, List`1 modulePathList)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Dictionary`2 functionsToDefine, Collection`1 errors, List`1 modulePathList, Boolean cacheResourcesFromMultipleModuleVersions)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Collection`1 errors)
   at System.Management.Automation.Language.Parser.ConfigurationStatementRule(IEnumerable`1 customAttributes, Token configurationToken) at line 7 column 15.
At line:1 char:1
+ Invoke-ScriptAnalyzer ./OctopusDSC/Examples/cOctopusSeqLogger.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ParserError: (ParserError:String) [Invoke-ScriptAnalyzer], ParseException
+ FullyQualifiedErrorId : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  System.DllNotFoundException: Unable to load DLL 'libmi.so': The specified module or one of its dependencies could not be found.
 (Exception from HRESULT: 0x8007007E)
   at Microsoft.Management.Infrastructure.Native.NativeMethods.MI_Application_InitializeV1(UInt32 flags, String applicationID, IndirectPtr extendedError, DirectPtr application)
   at Microsoft.Management.Infrastructure.Native.MI_Application.Initialize(String applicationId, MI_Instance& extendedError, MI_Application& application)
   at Microsoft.Management.Infrastructure.Internal.CimApplication.GetApplicationHandle()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.CreateValue()
   at Microsoft.Management.Infrastructure.Internal.CimApplication.get_Handle()
   at Microsoft.Management.Infrastructure.Serialization.CimMofDeserializer..ctor()
   at Microsoft.Management.Infrastructure.Serialization.CimMofDeserializer.Create()
   at Microsoft.PowerShell.DesiredStateConfiguration.CimDSCParser..ctor(OnClassNeeded onClassNeeded)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.ImportClasses(String path, Tuple`2 moduleInfo, Collection`1 errors)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.Initialize(Collection`1 errors, List`1 modulePathList)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Dictionary`2 functionsToDefine, Collection`1 errors, List`1 modulePathList, Boolean cacheResourcesFromMultipleModuleVersions)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Collection`1 errors)
   at System.Management.Automation.Language.Parser.ConfigurationStatementRule(IEnumerable`1 customAttributes, Token configurationToken) at line 7 column 15.,Microsoft.Windows.PowerShell.ScriptAnalyzer.Commands.InvokeScriptAnalyzerCommand
 
Invoke-ScriptAnalyzer : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  Unexpected token '{' in expression or statement at line 8 column 1.
At line:1 char:1
+ Invoke-ScriptAnalyzer ./OctopusDSC/Examples/cOctopusSeqLogger.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ParserError: (UnexpectedToken:String) [Invoke-ScriptAnalyzer], ParseException
+ FullyQualifiedErrorId : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  Unexpected token '{' in expression or statement at line 8 column 1.,Microsoft.Windows.PowerShell.ScriptAnalyzer.Commands.InvokeScriptAnalyzerCommand
 
Invoke-ScriptAnalyzer : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  Unexpected token '}' in expression or statement at line 22 column 1.
At line:1 char:1
+ Invoke-ScriptAnalyzer ./OctopusDSC/Examples/cOctopusSeqLogger.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ParserError: (UnexpectedToken:String) [Invoke-ScriptAnalyzer], ParseException
+ FullyQualifiedErrorId : Parse error in file /Users/me/OctopusDSC/OctopusDSC/Examples/cOctopusSeqLogger.ps1:  Unexpected token '}' in expression or statement at line 22 column 1.,Microsoft.Windows.PowerShell.ScriptAnalyzer.Commands.InvokeScriptAnalyzerCommand
 

```

Environment data
----------------

```powershell
> $PSVersionTable
 
Name                           Value                                                                                                                                                           
----                           -----                                                                                                                                                           
PSVersion                      6.0.0-rc.2                                                                                                                                                      
PSEdition                      Core                                                                                                                                                            
GitCommitId                    v6.0.0-rc.2                                                                                                                                                     
OS                             Darwin 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64                                                   
Platform                       Unix                                                                                                                                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                         
PSRemotingProtocolVersion      2.3                                                                                                                                                             
SerializationVersion           1.1.0.1                                                                                                                                                         
WSManStackVersion              3.0  
```

**Comment by bergmeister** [comment]
@matt-richardson This issue should be opened in the [PSScriptAnalzer](https://github.com/PowerShell/PSScriptAnalyzer) repo.p

**Comment by matt-richardson** [comment]
It appears to be an underlying issue, rather than related to PSScriptAnalyzer. Most likely an issue loading DSC. (I was short cutting the process of logging it over there and then getting told it should be logged over here.)

**Comment by iSazonov** [comment]
It is not supported.

**Comment by matt-richardson** [comment]
Wow. 3 years of waiting, and then I get 4 words. 
You ask for detailed information when someone files an issue, but then you close without any kind of response.

I dont want to use libmi. I want to use PSScriptAnalyzer on a DSC configuration file on macOS.

**Comment by iSazonov** [comment]
@matt-richardson It is old history and a plan is to exclude all dependencies on libmi.  MSFT team has started new cross platform DSC support project. See summary of the PR #13399.

**Comment by bergmeister** [comment]
@matt-richardson Last year, I improved PSScriptAnalyzer to be able to analyze DSC dependencies on non-Windows Unix systems as well when using the `-SaveDscDependency` switch in PR https://github.com/PowerShell/PSScriptAnalyzer/pull/1246, which was first released in PSScriptAnalyzer 1.18.1. Can you please try again with the latest version of PScriptAnalyzer by running `Install-Module PSScriptAnalyzer -Force` and restarting the shell in case PSScriptAnalyzer was already loaded before installing the newer version.
